tests/core_tests/test_path_cache.py

```diff
@@ -861,6 +861,46 @@ def _test_concurrent(self):
         self.assertFalse(self._multiprocess_fail)
 
 
+class TestPathCacheGetLocationsFullSync(TankTestBase):
+    """
+    Tests that Getting FilesystemLocations in a full sync only returns ones belonging to the current project.
+    """
+
+    def setUp(self):
+        super(TestPathCacheGetLocationsFullSync, self).setUp()
+
+        # Create a new project, we will assign a new Filesystemlocation entity to this
+        self._project_entity_b = self.mockgun.create("Project", {"name": "Project_B"})
+
+        # create a new FilesystemLocation entity and attach it to the default project 1 that generated by TankTestBase
+        # We hope to retrieve this entity in our test
+        self._asset_entity = self.mockgun.create("FilesystemLocation",
+                                                 {"code": "MyAsset_A", "project": {"type": "Project", "id":1}})
+
+        # create a new FilesystemLocation entity and associate it with the new project we created
+        # we hope not to retrieve this entity in our test
+        self._asset_entity = self.mockgun.create("FilesystemLocation",
+                                                 {"code": "MyAsset_B", "project": self._project_entity_b})
+
+        self._pc = path_cache.PathCache(self.tk)
+
+    def test_get_entities(self):
+        """
+        Check that we only get FilesystemLocation entities that belong to our project
+        """
+
+        # get the filesystemlocation entities as if it were a full sync.
+        entities = self._pc._get_filesystem_location_entities(folder_ids=None)
+        # we expect to only get back two entities. One is a default location entity created by TankTestBase
+        # and the other is the one we created and applied to project id 1 in the setup.
+        # we should not get back the other entity that was associated with project id 2
+        for entity in entities:
+            # now re find the entity returned, so that we can get back the project field and check it.
+            entity_w_project = self.tk.shotgun.find_one(path_cache.SHOTGUN_ENTITY,
+                                                        [["id","is", entity["id"]]],
+                                                        ["project"])
+            self.assertTrue(entity_w_project['project']['id'] == self.tk.pipeline_configuration.get_project_id())
```

Comment:
assertEqual would also work. It will print out both sides of the `==` sign in the console.